cmd/launcher/desktop.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/env"
+	"github.com/kolide/kit/logutil"
+	"github.com/kolide/launcher/ee/desktop"
+	"github.com/kolide/launcher/ee/desktop/menu"
+	"github.com/kolide/launcher/ee/desktop/server"
+	"github.com/oklog/run"
+	"github.com/shirou/gopsutil/process"
+)
+
+func runDesktop(args []string) error {
+	var (
+		flagset    = flag.NewFlagSet("kolide desktop", flag.ExitOnError)
+		flhostname = flagset.String(
+			"hostname",
+			"",
+			"hostname launcher is connected to",
+		)
+		flauthtoken = flagset.String(
+			"authtoken",
+			"",
+			"auth token for desktop server",
+		)
+	)
+
+	if err := setFlags(*flagset, args); err != nil {
+		return fmt.Errorf("setting flags: %w", err)
+	}
+
+	logger := logutil.NewServerLogger(env.Bool("LAUNCHER_DEBUG", false))
+	logger = log.With(logger, "component", "desktop_process")
+	level.Info(logger).Log("msg", "starting")
+
+	shutdownChan := make(chan struct{})
+
+	go handleSignals(logger, shutdownChan)
+	go monitorParentProcess(logger, shutdownChan)
+
+	var runGroup run.Group
+
+	server, err := server.New(logger, *flauthtoken, desktop.DesktopSocketPath(os.Getpid()), shutdownChan)
```

Comment:
I think I'm still very unsure about socket path creation. I _suspect_ we should have launcher create them, and pass them via an arg.

I'm not wholly sure why I think this. Just an instinct that this sort of assumption about paths and determinism, usually fails somewhere. (For example, on windows, we moved to using a random path for the thrift socket, because sometimes the old path was weirdly stale)

Reply:
finally working now to address this for windows we can just do the same as we do for launcher with some random string on the end of the socket file name

for the users should we create a specific folder for the user in the launcher root dir and give the user privs, then they can create the socket there? 

might look something like:
```
├── launcherRoot
│   ├── desktop_501
│   │   └── desktop.sock
│   └── desktop_502
│       └── desktop.sock
```